lib/active_model/serializer/adapter/json_api.rb

```diff
@@ -44,19 +44,15 @@ def fragment_cache(cached_hash, non_cached_hash)
 
         private
 
-        def add_relationships(resource, name, serializers)
-          resource[:relationships] ||= {}
-          resource[:relationships][name] ||= { data: [] }
-          resource[:relationships][name][:data] += serializers.map { |serializer| { type: serializer.json_api_type, id: serializer.id.to_s } }
-        end
-
-        def add_relationship(resource, name, serializer, val=nil)
-          resource[:relationships] ||= {}
-          resource[:relationships][name] = { data: val }
-
-          if serializer && serializer.object
-            resource[:relationships][name][:data] = { type: serializer.json_api_type, id: serializer.id.to_s }
-          end
+        def resource_identifier(serializer)
+          type = if ActiveModel::Serializer.config.jsonapi_resource_type == :plural
+                   serializer.object.class.model_name.plural
+                 else
+                   serializer.object.class.model_name.singular
+                 end
+          id = serializer.object.id.to_s
```

Comment:
This might be easier to read as a case statement

```ruby
type = 
  case
  when  ActiveModel::Serializer.config.jsonapi_resource_type == :plural
    serializer.object.class.model_name.plural
  else
    serializer.object.class.model_name.singular
  end
```

maybe `resource_type` and `resource_id` as the local variable names?  

This method seems like there's a latent object in here that transforms a single resource into a serializable hash

Nice on the naming  http://jsonapi.org/format/#document-resource-identifier-objects :)